backend/dataall/modules/dataset_sharing/services/data_sharing_service.py

```diff
@@ -111,50 +111,70 @@ def approve_share(cls, engine: Engine, share_uri: str) -> bool:
                     share_object_SM.update_state(session, share, new_object_state)
                     return False
 
```

Comment:
Maybe we can place the verification of the principal role before the dataset lock. In line 79 `lock_acquired = cls.acquire_lock_with_retry(dataset.datasetUri, session, share.shareUri)` we might be waiting and locking the dataset in vain for a share that is going to fail either way

Reply:
The lock is for the dataset resources, we want to prevent that the Bucket and KMS policies being updated during a share task are being updated simultaneously by another share task. A share item is unique for each share, there is no way another share is simultaneously updating it, plus RDS supports ACID transactions no?